src/geometry/collider_impl.rs

```diff
@@ -518,18 +518,8 @@ impl Collider {
     /// with a non-uniform scale results in an ellipse which isn’t supported),
     /// the shape is approximated by a convex polygon/convex polyhedron using
     /// `num_subdivisions` subdivisions.
-    pub fn set_scale(&mut self, mut scale: Vect, num_subdivisions: u32) {
-        /// We restrict the scaling increment to 1.0e-4, to avoid numerical jitter
-        /// due to the extraction of scaling factor from the GlobalTransform matrix.
-        fn snap_value(new: &mut f32) {
-            const PRECISION: f32 = 1.0e4;
-            *new = (*new * PRECISION).round() / PRECISION;
-        }
-
-        snap_value(&mut scale.x);
-        snap_value(&mut scale.y);
-        #[cfg(feature = "dim3")]
-        snap_value(&mut scale.z);
+    pub fn set_scale(&mut self, scale: Vect, num_subdivisions: u32) {
+        let scale = get_snapped_scale(scale);
 
         if scale == self.scale {
```

Comment:
What I have yet to figure out is why this doesn't work as expected.  I think something with how the the `&mut self` taking the mutable reference, triggering the Bevy change detector?

Reply:
Indeed, this is the case.  The `&mut self` will trigger a Bevy change every time.  I think it will be worth taking a closer look at some of the method impls on Rapier Components.


A simple reproduction:

```rust
#[derive(Component)]
pub struct Changeling {
    pub value: i32,
}

impl Changeling {
    pub fn increase_if_equal(&mut self, value: i32) {
        if self.value == value {
            self.value += 1;
        }
    }
}

pub fn print_changed(changes: Query<(Entity, &Changeling), Changed<Changeling>>) {
    for (e, change) in changes.iter() {
        log::info!("Entity changed! {:?} {:?}", e, change.value);
    }
}

pub fn change_once(mut changelings: Query<&mut Changeling>) {
    for mut cl in changelings.iter_mut() {
        cl.increase_if_equal(0);
    }
}
```

In the future, some updates will likely be able to be better handled with set_if_neq, which wsa added earlier this week:

https://github.com/bevyengine/bevy/blob/4820917af6be06866f10d4bd17191802d6640efb/crates/bevy_ecs/src/change_detection.rs